tests/unit/ui/components/TestOverlay.js

```diff
@@ -1,73 +1,141 @@
 
   it('renders extra className if provided', () => {
-    const root = render({ className: 'I-am-so-over-it' });
-    expect(root.overlayContainer.className).toContain('Overlay');
-    expect(root.overlayContainer.className).toContain('I-am-so-over-it');
+    const className = 'I-am-so-over-it';
+    const root = render({ className });
+    expect(root).toHaveClassName(className);
   });
 
   it('renders children', () => {
-    const root = render({ children: 'hello' });
-    expect(root.overlayContents).toBeTruthy();
-    expect(root.overlayContents.textContent).toContain('hello');
+    const html = '<div>overriding child div</div>';
+    const root = render({ children: html });
+    expect(
+      root
+        .children()
+        .at(1)
+        .props().children,
+    ).toEqual(html);
   });
 
   it('is hidden by default', () => {
     const root = render();
-    expect(root.overlayContainer.className).not.toContain('Overlay--visible');
+    expect(root).not.toHaveClassName('Overlay--visible');
   });
 
   it('is visible when the `visibleOnLoad` prop is passed', () => {
-    const root = render({ visibleOnLoad: true });
-    expect(root.overlayContainer.className).toContain('Overlay--visible');
+    const { store } = dispatchClientMetadata();
+
+    const root = render({
+      store,
+      visibleOnLoad: true,
+      uiState: { visible: false },
+    });
+
+    expect(root).not.toHaveClassName('Overlay--visible');
+
+    applyUIStateChanges({ root, store });
+
+    // Applying this 2 times here to mimic how uiState works in componentWillReceiveProps:
+    // The first time through, the visible state gets changed but these changes
+    // aren't shown in uiState till the 2 time through
+    applyUIStateChanges({ root, store });
+
+    expect(root).toHaveClassName('Overlay--visible');
   });
 
   it('hides when you click the background', () => {
-    const root = render({ visibleOnLoad: true });
-    Simulate.click(root.overlayBackground);
-    expect(root.overlayContainer.className).not.toContain('Overlay--visible');
+    const onClickBackgroundSpy = sinon.stub();
+    const root = render({
+      visibleOnLoad: true,
+      _onClickBackground: onClickBackgroundSpy,
+    });
+    const btn = root.find('.Overlay-background');
+    btn.simulate('click', createFakeEvent());
+    sinon.assert.called(onClickBackgroundSpy);
+    expect(root).not.toHaveClassName('Overlay--visible');
   });
 
   it('calls onEscapeOverlay when clicking the background', () => {
+    const clickEvent = createFakeEvent();
     const onEscapeOverlay = sinon.stub();
-    const root = render({ visibleOnLoad: true, onEscapeOverlay });
-    Simulate.click(root.overlayBackground);
+    const root = render({
+      visibleOnLoad: true,
+      onEscapeOverlay,
+    });
+    const btn = root.find('.Overlay-background');
+    btn.simulate('click', clickEvent);
+    const rootInstance = root.instance();
+    rootInstance.props._onClickBackground(clickEvent, rootInstance);
```

Comment:
Why do you need to invoke the method? Clicking on `'.Overlay-background'` should be enough -- was it not working in the test?

Reply:
hmm it is actually working. I'll remove..